### PR TITLE
fix: Implement intelligent touch passthrough for DivKit overlays

### PR DESCRIPTION
**Pull request by wankdanker**
Add content-aware touch handling to ActivationPreview that only intercepts touches on actual DivKit content, allowing touches on empty areas to pass through to underlying views (e.g., video player controls).

Changes:
- Add intelligent hit testing logic to detect DivKit content vs empty space
- Replace blanket OnClickListener with content-specific callback mechanism
- Implement onInterceptTouchEvent for precise touch event management
- Update demo JSON positioning and add notification permissions

[Screencast_20250711_210144.webm](https://github.com/user-attachments/assets/6a708733-ac02-4d36-bc7e-cb6a1d4a553d)

### CODE REVIEW

File: sourcesync-sdk-ui/src/main/java/io/sourcesync/sdk/ui/divkit/ActivationPreview.kt [hunk 1]
```diff
@@ -38,9 +42,101 @@ class ActivationPreview
...
-        // Add content container to frame layout
         addView(divView)
     }
+    
+    private fun setupTouchHandling() {
+        setOnTouchListener { _, event ->
+            when (event.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    // Check if touch hits actual DivKit content
+                    val hitView = findViewAtPoint(divView, event.x, event.y)
```
[the remark this answers is not in the document]
Comment (aaeb) [reply sourcesync-sdk-ui/src/main/java/io/sourcesync/sdk/ui/divkit/ActivationPreview.kt:60]: I appreciate your efforts on this, but as previously explained, there’s a more optimal solution that leverages Android’s rendering mechanism. Therefore, I’ll be closing this PR.
